uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/absence-tab-entitlements.component.js

```diff
@@ -11,16 +12,53 @@ define([
       return settings.pathTpl + 'components/absence-tab-entitlements.html';
     }],
     controllerAs: 'entitlements',
-    controller: ['$log', controller]
+    controller: AbsenceTabEntitlementsController
   });
 
-  function controller ($log) {
+  AbsenceTabEntitlementsController.$inject = ['$q', '$log', 'AbsenceType',
+    'OptionGroup'];
+
+  function AbsenceTabEntitlementsController ($q, $log, AbsenceType, OptionGroup) {
     $log.debug('Component: absence-tab-entitlements');
 
-    var vm = {};
+    var calculationUnits = [];
+    var vm = this;
+
+    vm.absenceTypes = [];
+    vm.loading = { component: true };
+
+    (function init () {
+      $q.all([
+        loadAbsenceTypes(),
+        loadCalculationUnits()
+      ])
+      .then(mapAbsenceTypeUnits)
+      .finally(function () {
+        vm.loading.component = false;
+      });
+    })();
+
+    function loadAbsenceTypes () {
+      return AbsenceType.all().then(function (data) {
+        vm.absenceTypes = data;
+      });
+    }
+
+    function loadCalculationUnits () {
```

Comment:
Please provide a description to this function (!) if this function is needed (see comments above)